### PR TITLE
C-functions to measure the time (user/sys/wallclock)

### PR DESCRIPTION
**Pull request by jposwiata**
to allow "profiling" the tests from the code (without additional instrumentation).

### CODE REVIEW

File: framework/sandstone.cpp [hunk 1]
```diff
@@ -660,6 +660,37 @@ inline void test_the_test_data<true>::test_tests_finish(const struct test *the_t
...
+    if (getrusage(RUSAGE_THREAD, &time) < 0) {
+        return 0;
+    }
+    return (1000000LL * time.ru_stime.tv_sec) + time.ru_stime.tv_usec - from;
+#else
+    return 0;
+#endif
+}
+
+uint64_t sandstone_wall_clock_time(uint64_t from)
```
Comment (lukasz-tuz) [inline framework/sandstone.cpp:689]: I think this should be moved to `sandstone_chrono.h`.

File: framework/sandstone.cpp [hunk 2]
```diff
@@ -660,6 +660,37 @@ inline void test_the_test_data<true>::test_tests_finish(const struct test *the_t
 #undef maybe_log_error
 }
 
+uint64_t sandstone_user_cpu_time(uint64_t from)
```
Comment (lukasz-tuz) [inline framework/sandstone.cpp:663]: `sandstone_test_utils.h` feels like a better home for these two functions (`sandstone_user_cpu_time` and `sandstone_sys_cpu_time`). Also, please add unit tests for these.

**Review by lukasz-tuz — DISMISSED** [review]
Is there any benefit for using these functions over a profiler (perf, valgrind, vtune)?

File: framework/sandstone.cpp (hunk 1 above)
Comment (jposwiata) [reply framework/sandstone.cpp:689]: 'chrono' sounds good.

**Comment by jposwiata** [comment]
> Is there any benefit for using these functions over a profiler (perf, valgrind, vtune)?

wall-clock time allows to measure overall processing (with much higher granularity than "thread time"), e.g. allows to measure time of handling a message with inter-thread communication, measure the time without framework overhead, etc.
Cpu time is fully covered by perf/vtune (with much better resolution), and therefore these functions are intended to get "first impression" only, by execution the app without involving mentioned (heavy and difficult to use) tools.

**Review by lukasz-tuz — APPROVED** [review]

**Comment by thiagomacieira** [comment]
```
[ RUN      ] SandstoneTime.CpuTime
../opendcdiag/framework/unit-tests/sandstone_utils_tests.cpp:73: Failure
Expected: (proc_sys + proc_user) <= (proc_wall), actual: 4262 vs 4246
```
Please fix.

**Pull request merged**